### PR TITLE
(PC-7940) add postal code sign up 

### PR DESCRIPTION
**Pull request by emmajmn**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-7940

## Checklist

I have:

- [x] Made sure the title of my PR follows the convention `[PC-XXX] <summary>`.
- [x] Made sure my feature is working on the relevant real / virtual devices.
- [x] Written **unit tests** for my feature.
- [ ] Added new reusable components to the AppComponents page and communicate to the team on slack
- [x] Added a **screenshot** for UI tickets.
- [ ] Attached a **ticket number** for any added TODO/FIXME \
       (for tech tasks, give the best context about the TODO resolution: what/who/when).

## Screenshots

| \*iOS arrivée sur la page code postal | iOS 5 chiffres max |
| -------------------- | :----------------------: | 
| image | image | 

## Deploy hard

If native code (ios/android) was modified, **after** the PR is merged, on the master branch, upgrade the **app version** (+1 patch):

- if you want an hard deployment of the testing environment, use `yarn trigger:testing:deploy`

### CODE REVIEW

File: src/features/auth/signup/SetPostalCode.tsx [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+  const {
+    visible: fullPageModalVisible,
+    showModal: showFullPageModal,
+    hideModal: hideFullPageModal,
+  } = useModal(false)
+
+  const email = route.params.email
+  const isNewsletterChecked = route.params.isNewsletterChecked
+  const password = route.params.password
+  const birthday = route.params.birthday
```
Comment (justinemignot) [inline src/features/auth/signup/SetPostalCode.tsx:40]: tu peux regrouper ces 4 lignes en une: 
const { email, isNewsletterChecked, password, birthday } = route.params 
:)

File: src/features/auth/signup/SetPostalCode.tsx [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    }
+  }
+
+  function showQuitSignupModal() {
+    postalCodeInput.current && postalCodeInput.current.blur()
+    showFullPageModal()
+  }
+
+  return (
+    <React.Fragment>
```
Comment (justinemignot) [inline src/features/auth/signup/SetPostalCode.tsx:63]: pourquoi a-t-on besoin de ce fragment ?

File: src/features/auth/signup/SetPostalCode.tsx [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+})
+
+const StyledInput = styled.View({
+  display: 'flex',
+  flexDirection: 'column',
+  alignItems: 'flex-start',
+  width: '100%',
+})
+
+const Paragraphe = styled.Text({
```
Comment (justinemignot) [inline src/features/auth/signup/SetPostalCode.tsx:138]: du français ici "Paragraphe" à traduire :)

File: src/features/auth/signup/SetPostalCode.tsx (hunk 1 above)
Comment (emmajmn) [reply src/features/auth/signup/SetPostalCode.tsx:40]: yes

File: src/features/auth/signup/SetPostalCode.tsx (hunk 2 above)
Comment (emmajmn) [reply src/features/auth/signup/SetPostalCode.tsx:63]: car il entoure le bottomContentPage et le QuitSignUpModal (donc 2 composants enfant du fragment)

**Review by justinemignot — APPROVED** [review]

**Pull request merged**